pyomo/contrib/doe/DoE_package.py

```diff
@@ -0,0 +1,1812 @@
+        total_count = len(dv_ranges[self.dv_name[0]]) * len(dv_ranges[self.dv_name[1]])
+
+        # loop over design variables
+        for i, value_0 in enumerate(dv_ranges[self.dv_name[0]]):
+            for j, value_1 in enumerate(dv_ranges[self.dv_name[1]]):
+                # generate the design variable dictionary needed for running compute_FIM
+                design_iter = {}
+
+                # first copy value from design_Values
+                dv_iter1 = design_values[self.dv_name[0]].copy()
+                dv_iter2 = design_values[self.dv_name[1]].copy()
+
+                # for timepoints given in dv_apply_times, change its value to be the ones in dv_ranges to go over
+                for t, tim in enumerate(self.dv_control_times[0]):
+                    dv_iter1[tim] = value_0
+                design_iter[self.dv_name[0]] = dv_iter1
+
+                for t, tim in enumerate(self.dv_control_times[1]):
+                    dv_iter2[tim] = value_1
+                design_iter[self.dv_name[1]] = dv_iter2
+
+                print('=======This is the ', count+1, 'th iteration=======')
+                print('Design variable values of this iteration:', design_iter)
+
+                t_each_begin = time.time()
+
+                # call compute_FIM to get FIM
+                result_iter = self.compute_FIM(design_iter, mode=self.mode,
+                                               tee_opt=tee_option,
+                                               scale_nominal_param_value=self.scale_nominal_param_value,
+                                               formula=formula, step=step)
+
+                if (self.mode=='simultaneous_finite'):
+                    result_iter.extract_FIM(self.m, self.design_timeset, self.square_result, self.obj_opt)
+
+                elif (self.mode == 'sequential_finite'):
+                    result_iter.calculate_FIM(self.jac, self.design_values)
+
+                elif (self.mode == 'sequential_sipopt'):
+                    result_iter.calculate_FIM(self.jac, self.design_values)
+
+                t_now = time.time()
+
+                if self.verbose:
+                    # give run information at each iteration
+                    print('This is the ', count+1, ' run out of ', total_count, 'run.')
+                    print('The code has run %.04f seconds.'% (t_now-t_begin))
+                    print('Estimated remaining time: %.4f seconds' % ((t_now-t_begin)/(count+1)*(total_count-count-1)))
+                count += 1
+
+                result_combine.append(result_iter)
+
+        t_end = time.time()
+        print('The whole run takes ', t_end - t_begin, ' s.')
+
+        # store results
+        if self.filename is not None:
+            f = open(filename, 'wb')
+            pickle.dump(result_combine, f)
+            f.close()
+
+        return result_combine
+
+    def sensitivity_analysis_1D(self, design_values, design_var_range, sensitivity_step=0.1, compare_opt='D',
```

Comment:
Can we generalize the previous method to support 1, 2, or N dimensions?

Reply:
@adowling2 Do you mean we should generate the grid_search() function to support 1 dimension or more than 2 dimensions? 
I think the biggest question here is how we loop over the dimensions. Now for two dimensions, we use two for loops. Do you have suggestions on how should I automate these for loops? Or we just check how many dimensions are needed, and write separate code for 1 dimension, 2 dimension and 3 dimension?

Reply:
Look at using products in `itertools`. Here is another project in the group that uses it: https://github.com/dowlinglab/reactive-inks/blob/782316ec2977434add973c341796542132721860/reaction_models.py#L894